bin/daemon

```diff
@@ -29,17 +29,26 @@ end
 @checksum = calc_checksum
 
 module Updater
+  class UpdateFailed < StandardError
+  end
+
   extend Scarlet::Daemon::Loggable
 
   def self.git_update
     data = Scarlet::Git.get_data
     branch = data[:branch]
-    throw :update_failed, 'no branch' unless branch
-    system("git fetch origin") || throw(:update_failed, 'Fetching failed')
-    system("git reset --hard origin/#{branch}") || throw(:update_failed, 'Reset failed')
+    raise UpdateFailed, 'no branch obtained.' unless branch
+
+    system("git fetch origin") || raise(UpdateFailed, 'git fetch failed')
+    system("git reset --hard origin/#{branch}") || raise(UpdateFailed, 'git reset failed')
+
     current_data = Scarlet::Git.get_data
     log "Updated from #{data[:commit]} to #{current_data[:commit]}"
-    false # false as in, did I fail?
+  end
+
+  def self.install_gems
+    log "Installing gems"
+    system("bundle install")
   end
 
   def self.git_rollback(commit, reason)
```

Comment:
That's the whole point. We don't change anything by using `checkout`, while `reset` modifies the actual branch. To update code, use `fetch`, and it should just work. For conflicts, there shouldn't be any, as we're only dealing with commited code, and in the other case I'd prefer to be alerted about changes to the codebase instead of silently deleting them.